src/scrapers/mizrahi.js

```diff
@@ -0,0 +1,155 @@
+    { selector: '#ctl00_PlaceHolderLogin_ctl00_tbPassword', value: credentials.password },
+  ];
+}
+
+function getPossibleLoginResults() {
+  const urls = {};
+  urls[LOGIN_RESULT.SUCCESS] = [AFTER_LOGIN_BASE_URL];
+  urls[LOGIN_RESULT.INVALID_PASSWORD] = [`${BASE_URL}/login/loginMTO.aspx`];
+  urls[LOGIN_RESULT.CHANGE_PASSWORD] = [
+    `${AFTER_LOGIN_BASE_URL}/main/uis/ge/changePassword/`,
+  ];
+  return urls;
+}
+
+function CreateDataFromRequest(request, optionsStartDate) {
+  const defaultStartMoment = moment().subtract(1, 'years');
+  const startDate = optionsStartDate || defaultStartMoment.toDate();
+  const startMoment = moment.max(defaultStartMoment, moment(startDate));
+
+  const data = JSON.parse(request.postData());
+
+  data.inToDate = moment().format(DATE_FORMAT);
+  data.inFromDate = startMoment.format(DATE_FORMAT);
+  data.table.maxRow = MAX_ROWS_PER_REQUEST;
+
+  return data;
+}
+
+function createHeadersFromRequest(request) {
+  return {
+    mizrahixsrftoken: request.headers().mizrahixsrftoken,
+    'Content-Type': request.headers()['content-type'],
+  };
+}
+
+function convertTransactions(txns) {
+  return txns.map((row) => {
+    // row.MC02PeulaTaaEZ: 2019-09-10T00:00:00
+    const txnDate = moment(row.MC02PeulaTaaEZ, moment.HTML5_FMT.DATETIME_LOCAL_SECONDS)
+      .format(ISO_DATE_FORMAT);
+
+    return {
+      type: NORMAL_TXN_TYPE,
+      identifier: row.MC02AsmahtaMekoritEZ ? parseInt(row.MC02AsmahtaMekoritEZ, 10) : null,
+      date: txnDate,
+      processedDate: txnDate,
+      originalAmount: row.MC02SchumEZ,
+      originalCurrency: SHEKEL_CURRENCY,
+      chargedAmount: row.MC02SchumEZ,
+      description: row.MC02TnuaTeurEZ,
+      status: TRANSACTION_STATUS.COMPLETED,
+    };
+  });
+}
+
+async function extractPendingTransactions(page) {
+  const pendingTxn = await pageEvalAll(page, 'tr.rgRow', [], (trs) => {
+    return trs.map((tr) => Array.from(tr.querySelectorAll('td'), (td) => td.textContent));
+  });
+
+  return pendingTxn.map((txn) => {
+    const date = moment(txn[0], 'DD/MM/YY').format(ISO_DATE_FORMAT);
+    const amount = parseInt(txn[3], 10);
+    return {
+      type: NORMAL_TXN_TYPE,
+      identifier: null,
+      date,
+      processedDate: date,
+      originalAmount: amount,
+      originalCurrency: SHEKEL_CURRENCY,
+      chargedAmount: amount,
+      description: txn[1],
+      status: TRANSACTION_STATUS.PENDING,
+    };
+  });
+}
+
+class MizrahiScraper extends BaseScraperWithBrowser {
+  async sniffOshRequest() {
+    await waitForNavigation(this.page);
+    this.navigateTo(OSH_PAGE, this.page);
+    this.request = await this.page.waitForRequest(TRANSACTIONS_REQUEST_URL);
+  }
+
+  getLoginOptions(credentials) {
+    return {
+      loginUrl: `${LOGIN_URL}`,
+      fields: createLoginFields(credentials),
+      submitButtonSelector: '#ctl00_PlaceHolderLogin_ctl00_Enter',
+      postAction: async () => this.sniffOshRequest(),
```

Comment:
@baruchiro I couldn't figure out why we're sniffing osh request during the `postAction`. Couldn't we simply move everything inside the `sniffOshRequest()` function to the beginning of the `fetchData()` function?
that way we don't need to have a `request` field in the class (unless I'm missing something).

Reply:
Resolved.
But there is a workaround for #277 .